packages/xrpl/tools/generateModels.js

```diff
@@ -0,0 +1,265 @@
+  )
+  const sfields = {}
+  for (const hit of sfieldHits) {
+    const matches = hit.match(
+      /^ *CONSTRUCT_[^\_]+_SFIELD *\( *[^,\n]*,[ \n]*"([^\"\n ]+)"[ \n]*,[ \n]*([^, \n]+)[ \n]*,[ \n]*([0-9]+)(,.*?(notSigning))?/,
+    )
+    sfields[matches[1]] = matches.slice(2)
+  }
+
+  const txFormatsCpp = readFile(
+    path.join(folder, 'src/ripple/protocol/impl/TxFormats.cpp'),
+  )
+  const txFormatsHits = txFormatsCpp.match(
+    /^ *add\(jss::([^\"\n, ]+),[ \n]*tt[A-Z_]+,[ \n]*{[ \n]*(({sf[A-Za-z0-9]+, soe(OPTIONAL|REQUIRED|DEFAULT)},[ \n]+)*)},[ \n]*[pseudocC]+ommonFields\);/gm,
+  )
+  const txFormats = {}
+  for (const hit of txFormatsHits) {
+    const matches = hit.match(
+      /^ *add\(jss::([^\"\n, ]+),[ \n]*tt[A-Z_]+,[ \n]*{[ \n]*(({sf[A-Za-z0-9]+, soe(OPTIONAL|REQUIRED|DEFAULT)},[ \n]+)*)},[ \n]*[pseudocC]+ommonFields\);/,
+    )
+    txFormats[matches[1]] = formatTxFormat(matches[2])
+  }
+
+  jsTransactionFile = readFile(
+    path.join(
+      path.dirname(__filename),
+      '../src/models/transactions/transaction.ts',
+    ),
+  )
+  const transactionMatch = jsTransactionFile.match(
+    /export type Transaction =([| \nA-Za-z]+)\nexport/,
+  )[0]
+  const existingLibraryTxs = transactionMatch
+    .replace('\n\nexport', '')
+    .split('\n  | ')
+    .filter((value) => !value.includes('export type'))
+    .map((value) => value.trim())
+  existingLibraryTxs.push('EnableAmendment', 'SetFee', 'UNLModify')
+
+  const txsToAdd = []
+
+  for (const tx in txFormats) {
+    if (!existingLibraryTxs.includes(tx)) {
+      txsToAdd.push(tx)
+    }
+  }
+
+  return [txsToAdd, txFormats, sfields, transactionMatch]
+}
+
+function formatTxFormat(rawTxFormat) {
+  return rawTxFormat
+    .trim()
+    .split('\n')
+    .map((element) => element.trim().replace(/[{},]/g, '').split(' '))
+}
+
+const typeMap = {
+  UINT8: 'number',
+  UINT16: 'number',
+  UINT32: 'number',
+  UINT64: 'number | string',
+  UINT128: 'string',
+  UINT160: 'string',
+  UINT256: 'string',
+  AMOUNT: 'Amount',
+  VL: 'string',
+  ACCOUNT: 'string',
+  VECTOR256: 'string[]',
+  PATHSET: 'Path[]',
+  ISSUE: 'Currency',
+  XCHAIN_BRIDGE: 'XChainBridge',
+  OBJECT: 'any',
+  ARRAY: 'any[]',
+}
+
+const allCommonImports = ['Amount', 'Currency', 'Path', 'XChainBridge']
+const additionalValidationImports = ['string', 'number']
+
+function updateTransactionFile(transactionMatch, tx) {
+  const transactionMatchSplit = transactionMatch.split('\n  | ')
+  const firstLine = transactionMatchSplit[0]
+  const allTransactions = transactionMatchSplit.slice(1)
+  allTransactions.push(tx)
+  allTransactions.sort()
+  const newTransactionMatch =
+    firstLine + '\n  | ' + allTransactions.join('\n  | ')
+  let newJsTxFile = jsTransactionFile.replace(
+    transactionMatch,
+    newTransactionMatch,
+  )
+
+  newJsTxFile = newJsTxFile.replace(
+    `import {
```

Comment:
This feels oddly specific to one type.

Reply:
It's the last type in the list, it was easier to do it this way and have the linter fix the sorting than do some complex regex that sorts alphabetically.

Reply:
Can you add a comment with that context?